test/format/config.c

```diff
@@ -766,75 +766,95 @@ config_pct(void)
 static void
 config_transaction(void)
 {
-    bool prepare_requires_ts;
-
     /*
      * WiredTiger cannot support relaxed isolation levels. Turn off everything but timestamps with
      * snapshot isolation.
      */
     if ((!g.c_txn_timestamps && config_is_perm("transaction.timestamps")) ||
       (g.c_isolation_flag != ISOLATION_SNAPSHOT && config_is_perm("transaction.isolation")))
-        testutil_die(EINVAL, "format limited to timestamp and snapshot-isolation runs");
+        testutil_die(EINVAL, "format limited to timestamp and snapshot-isolation testing");
+    if (!g.c_txn_timestamps)
+        config_single("transaction.timestamps=on", false);
+    if (g.c_isolation_flag != ISOLATION_SNAPSHOT)
+        config_single("transaction.isolation=snapshot", false);
 
     /*
-     * We can't prepare a transaction if logging is configured or timestamps aren't configured.
-     * Further, for repeatable reads to work in timestamp testing, all updates must be within a
-     * snapshot-isolation transaction. Check for incompatible configurations, then let prepare and
-     * timestamp drive the remaining configuration.
+     * Check the permanent configuration. We can't prepare a transaction if logging is configured or
+     * timestamps aren't configured. For repeatable reads to work in timestamp testing, all updates
+     * must be done in a snapshot isolation transaction.
      */
-    prepare_requires_ts = false;
-    if (g.c_prepare) {
-        if (config_is_perm("ops.prepare")) {
-            if (g.c_logging && config_is_perm("logging"))
-                testutil_die(EINVAL, "prepare is incompatible with logging");
-            if (!g.c_txn_timestamps && config_is_perm("transaction.timestamps"))
-                testutil_die(EINVAL, "prepare requires transaction timestamps");
-        } else if ((g.c_logging && config_is_perm("logging")) ||
-          (!g.c_txn_timestamps && config_is_perm("transaction.timestamps")))
-            config_single("ops.prepare=off", false);
-        if (g.c_prepare) {
-            prepare_requires_ts = true;
-            if (g.c_logging)
-                config_single("logging=off", false);
-            if (!g.c_txn_timestamps)
-                config_single("transaction.timestamps=on", false);
-        }
+    if (g.c_prepare && config_is_perm("ops.prepare")) {
+        if (g.c_logging && config_is_perm("logging"))
+            testutil_die(EINVAL, "prepare is incompatible with logging");
+        if (!g.c_txn_timestamps && config_is_perm("transaction.timestamps"))
+            testutil_die(EINVAL, "prepare requires transaction timestamps");
+        if (g.c_isolation_flag != ISOLATION_SNAPSHOT && config_is_perm("transaction.isolation"))
+            testutil_die(EINVAL, "prepare requires snapshot isolation");
+        if (g.c_txn_freq != 100 && config_is_perm("transaction.frequency"))
+            testutil_die(EINVAL, "prepare requires transaction frequency set to 100");
```

Comment:
Where did this requirement come from?

Reply:
Not sure what you're asking?

Reply:
I don't see the frequency=100 in the "before" code.

Reply:
Oh, I added some stuff that was missing as part of review. Basically, timestamps require everything to be inside of a transction, atm.